pkg/kubectl/cmd/taint.go

```diff
@@ -0,0 +1,386 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"encoding/json"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/util/strategicpatch"
+	"k8s.io/kubernetes/pkg/util/validation"
+)
+
+// TaintOptions have the data required to perform the annotate operation
+type TaintOptions struct {
+	resources       []string
+	taintsToAdd     []api.Taint
+	removeTaintKeys []string
+	builder         *resource.Builder
+	selector        string
+	overwrite       bool
+	all             bool
+	f               *cmdutil.Factory
+	out             io.Writer
+	cmd             *cobra.Command
+}
+
+const (
+	taint_long = `Update the taints on a or more nodes.
+
+A taint must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to %[1]d characters.`
```

Comment:
A taint is a key + value + effect, not a string, so this sentence is confusing. You could say something like "A taint consists of a key, value, and effect. As an argument here, it is expressed as key=value:effect, where the key must <...>, the value must <...>, and the effect must be NoSchedule or PreferNoSchedule" (please add a comment to update this when we add other effects). Does that make sense?